pkg/analysis/ssatags/testdata/src/a/a.go

```diff
@@ -0,0 +1,27 @@
+package a
+
+type SSATagsSpec struct {
+	// +kubebuilder:listType=atomic
+	AtomicList []string `json:"atomicList,omitempty"`
+
+	// +kubebuilder:listType=set
+	SetList []string `json:"setList,omitempty"` // want "listType=set is forbidden, use listType=atomic or listType=map instead"
+
+	// +kubebuilder:listType=map
+	// +kubebuilder:listMapKey=name
```

Comment:
I wonder if we should check that `map` is only on list of objects, and that the key actually exists?

Reply:
It's really really good. I'd implement it 👍